pkg/cmd/enable/addons/exec.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Contributors to the Open Cluster Management project
+package addons
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/cmd/enable/addons/scenario"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+)
+
+//ClusterName: The cluster name used in the template
+type ClusterName struct {
+	ClusterName string
+}
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("addon options:", "dry-run", o.ClusteradmFlags.DryRun, "names", o.names, "clusters", o.clusters, "output-file", o.outputFile)
+
+	return nil
+}
+
+func (o *Options) validate() error {
+	if o.names == "" {
+		return fmt.Errorf("names is missing")
+	}
+
+	names := strings.Split(o.names, ",")
+	for _, n := range names {
+		if n != "application-manager" {
+			return fmt.Errorf("invalid add-on name %s", n)
+		}
+	}
+
+	if o.clusters == "" {
+		return fmt.Errorf("clusters is misisng")
+	}
+
+	return nil
+}
+
+func (o *Options) run() error {
+	alreadyProvidedAddons := make(map[string]bool)
+	addons := make([]string, 0)
+	names := strings.Split(o.names, ",")
+	for _, n := range names {
+		if _, ok := alreadyProvidedAddons[n]; !ok {
+			alreadyProvidedAddons[n] = true
+			addons = append(addons, strings.TrimSpace(n))
+		}
+	}
+	o.values.addons = addons
+
+	alreadyProvidedClusters := make(map[string]bool)
+	clusters := make([]string, 0)
+	cs := strings.Split(o.clusters, ",")
+	for _, c := range cs {
+		if _, ok := alreadyProvidedClusters[c]; !ok {
+			alreadyProvidedClusters[c] = true
+			clusters = append(clusters, strings.TrimSpace(c))
+		}
+	}
+	o.values.clusters = clusters
+
+	klog.V(3).InfoS("values:", "addons", o.values.addons, "clusters", o.values.clusters)
+
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	clusterClient, err := clusterclientset.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	for _, clusterName := range o.values.clusters {
+		_, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
+			clusterName,
+			metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	output := make([]string, 0)
+	reader := scenario.GetScenarioResourcesReader()
+
+	kubeClient, apiExtensionsClient, dynamicClient, err := helpers.GetClients(o.ClusteradmFlags.KubectlFactory)
+	if err != nil {
+		return err
+	}
+
+	applierBuilder := &apply.ApplierBuilder{}
+	applier := applierBuilder.WithClient(kubeClient, apiExtensionsClient, dynamicClient).Build()
+
+	for _, addon := range o.values.addons {
+		if addon == "application-manager" {
```

Comment:
I think  we may not install hub  controller here, or we need  to explicitly call out somewhere (maybe in the flags or help message of cmd?) that application-manager is a built-in addon.

Also I would think we need to split "enable addon on managed cluster" and "install addon" to two commands

Reply:
done as recommended.